application/ingestr/kafka-cmd.sh

```diff
@@ -0,0 +1,199 @@
+  echo "Done."
+}
+
+function delete-topic() {
+  title "Deleting Kafka topic"
+  docker compose --file ${COMPOSE_FILE} run --rm --no-TTY delete-topic
+  echo "Done."
+}
+
+
+# --------
+# Pipeline
+# --------
+
+function invoke-job() {
+
+  # Invoke ingestr job.
+  title "Invoking ingestr job"
+
+  uvx --python="3.12" --prerelease="allow" --with-requirements="requirements.txt" ingestr ingest --yes \
+    --source-uri "kafka://?bootstrap_servers=localhost:${PORT_KAFKA_BROKER_EXTERNAL}&group_id=test_group" \
+    --source-table "demo" \
+    --dest-uri "cratedb://crate:crate@localhost:5432/?sslmode=disable" \
+    --dest-table "doc.kafka_demo"
+
+  echo "Done."
+}
+
+function feed-data() {
+
+  if [[ ! -f nyc-yellow-taxi-2017-subset.ndjson ]]; then
+
+    title "Acquiring NDJSON data"
+
+    # Acquire NYC Taxi 2017 dataset in JSON format (~90 MB).
+    wget --no-clobber --continue https://gist.githubusercontent.com/kovrus/328ba1b041dfbd89e55967291ba6e074/raw/7818724cb64a5d283db7f815737c9e198a22bee4/nyc-yellow-taxi-2017.tar.gz
+
+    # Extract archive.
+    tar -xvf nyc-yellow-taxi-2017.tar.gz
+
+    # Create a subset of the data (5000 records) for concluding the first steps.
+    cat nyc-yellow-taxi-2017.json | head -n 5000 > nyc-yellow-taxi-2017-subset.ndjson
+
+  fi
+
+  # Publish data to the Kafka topic.
+  title "Publishing NDJSON data to Kafka topic"
+  cat nyc-yellow-taxi-2017-subset.ndjson | docker compose --file ${COMPOSE_FILE} run --rm --no-TTY publish-data
+  echo "Done."
+
+  # Wait a bit for the data to transfer and converge successfully.
+  sleep 3
+}
+
+
+# ---------
+# Data sink
+# ---------
+
+function create-table() {
+  title "Creating CrateDB table"
+  docker compose --file ${COMPOSE_FILE} run --rm create-table
+  echo "Done."
+}
+
+function display-data() {
+
+  title "Displaying data in CrateDB"
+
+  docker compose --file ${COMPOSE_FILE} run --rm httpie \
+    http "${CRATEDB_HTTP_URL}/_sql?pretty" stmt='REFRESH TABLE "kafka_demo";' --ignore-stdin > /dev/null
+
+  docker compose --file ${COMPOSE_FILE} run --rm httpie \
+    http "${CRATEDB_HTTP_URL}/_sql?pretty" stmt='SELECT * FROM "kafka_demo" LIMIT 5;' --ignore-stdin
+
+  docker compose --file ${COMPOSE_FILE} run --rm httpie \
+    http "${CRATEDB_HTTP_URL}/_sql?pretty" stmt='SELECT COUNT(*) FROM "kafka_demo";' --ignore-stdin
+
+}
+
+function verify-data() {
+  title "Verifying data in CrateDB"
+  size_reference=5000
+  size_actual=$(
+    docker compose --file ${COMPOSE_FILE} run --rm httpie \
+      http "${CRATEDB_HTTP_URL}/_sql?pretty" stmt='SELECT COUNT(*) FROM "kafka_demo";' --ignore-stdin \
+    | jq .rows[0][0]
+  )
+  if [[ "${size_actual}" = "${size_reference}" ]]; then
+    echo -e "${BGREEN}SUCCESS: Database table contains expected number of ${size_reference} records.${NC}"
+  else
+    echo -e "${BRED}ERROR:   Expected database table to contain ${size_reference} records, but it contains ${size_actual} records.${NC}"
+    exit 2
+  fi
+}
```

Comment:
Backlog: Data validation is currently a bit thin, just counting the number of records and failing if it's not 5_000. Please expand the procedure, to also consider if data is in the right format within the target table. Also, why in hell is this written in Bash?